docs/source/training_tutorials/sft_lora_finetune_llm.mdx

```diff
@@ -0,0 +1,424 @@
+export MALLOC_ARENA_MAX=64
+export NEURON_CC_FLAGS="--model-type=transformer --distribution-strategy=llm-training --enable-saturate-infinity --cache_dir=/home/ubuntu/cache_dir_neuron/"
+
+PROCESSES_PER_NODE=8
+
+NUM_EPOCHS=1
+TP_DEGREE=2
+PP_DEGREE=1
+BS=1
+GRADIENT_ACCUMULATION_STEPS=8
+LOGGING_STEPS=1
+MODEL_NAME="meta-llama/Meta-Llama-3-8B"
+OUTPUT_DIR=output-$SLURM_JOB_ID
+
+if [ "$NEURON_EXTRACT_GRAPHS_ONLY" = "1" ]; then
+    MAX_STEPS=$((LOGGING_STEPS + 5))
+else
+    MAX_STEPS=-1
+fi
+
+
+XLA_USE_BF16=1 torchrun --nproc_per_node $PROCESSES_PER_NODE docs/source/training_tutorials/sft_lora_finetune_llm.py \
+  --model_id $MODEL_NAME \
+  --num_train_epochs $NUM_EPOCHS \
+  --do_train \
+  --learning_rate 5e-5 \
+  --warmup_ratio 0.03 \
+  --max_steps $MAX_STEPS \
+  --per_device_train_batch_size $BS \
+  --per_device_eval_batch_size $BS \
+  --gradient_accumulation_steps $GRADIENT_ACCUMULATION_STEPS \
+  --gradient_checkpointing true \
+  --bf16 \
+  --zero_1 false \
+  --tensor_parallel_size $TP_DEGREE \
+  --pipeline_parallel_size $PP_DEGREE \
+  --logging_steps $LOGGING_STEPS \
+  --save_total_limit 1 \
+  --output_dir $OUTPUT_DIR \
+  --lr_scheduler_type "constant" \
+  --overwrite_output_dir
+```
+
+That's it, we successfully trained Llama-3 8B on AWS Trainium!
+
+But before we can share and test our model we need to consolidate our model. Since we used tensor parallelism during training, we saved sharded versions of the checkpoints. We need to consolidate them now.
+
+### Consolidate the Checkpoint
+
+The Optimum CLI provides a way of doing that very easily via the `optimum neuron consolidate [sharded_checkpoint] [output_dir]` command:
+
+```bash
+optimum-cli neuron consolidate dolly_llama dolly_llama
+```
+
+## 5. Evaluate and test fine-tuned Llama model
+
+As for training, to be able to run inference on AWS Trainium or AWS Inferentia2 we need to compile our model. In this case, we will use our Trainium instance for the inference test, but we recommend customer to switch to Inferentia2 for inference.
+
+Optimum Neuron implements similar to Transformers AutoModel classes for easy inference use. We will use the `NeuronModelForCausalLM` class to load our vanilla transformers checkpoint and convert it to neuron.
+
+```python
+from optimum.neuron import NeuronModelForCausalLM
+from transformers import AutoTokenizer
+
+compiler_args = {"num_cores": 2, "auto_cast_type": 'fp16'}
+input_shapes = {"batch_size": 1, "sequence_length": 2048}
+
+tokenizer = AutoTokenizer.from_pretrained("dolly_llama")
+model = NeuronModelForCausalLM.from_pretrained(
+        "dolly_llama",
+        export=True,
+        **compiler_args,
+        **input_shapes)
+```
+
+_Note: Inference compilation can take ~25minutes. Luckily, you need to only run this onces. Since you can save the model afterwards. If you are going to run on Inferentia2 you need to recompile again. The compilation is parameter and hardware specific._
```

Comment:
Question, if we have precompiled and cache the official llama3 8B, will the precompiled artifacts be fetched from the cache repo for the tranium fine-tuned checkpoint or do we need to recompile everything? @dacorvo

Reply:
For inference in prod? It will recompile for sure.